code/game/machinery/iv_drip.dm

```diff
@@ -1,224 +1,684 @@
+			if(breather)
+				to_chat(usr, SPAN_NOTICE("You cannot remove \the [breath_mask] if someone's wearing it!"))
+				return
+			usr.visible_message(SPAN_NOTICE("[usr] removes \the [breath_mask] from \the [src]."), SPAN_NOTICE("You remove \the [breath_mask] from the \the [src]."))
+			breath_mask.forceMove(usr.loc)
+			usr.put_in_hands(breath_mask)
+			breath_mask = null
+			update_icon()
+		else
+			return
 
-	if(!istype(attached))
+/obj/machinery/iv_drip/attackby(obj/item/W as obj, mob/user as mob)
+	if(istype(W, /obj/item/reagent_containers/blood/ripped))
+		to_chat(user, "You can't use a ripped bloodpack.")
 		return
-
-	if(!(get_dist(src, attached) <= 1 && isturf(attached.loc)))
-		var/obj/item/organ/external/affecting = attached.get_organ(pick(BP_R_ARM, BP_L_ARM))
-		attached.visible_message("<span class='warning'>The needle is ripped out of [attached]'s [affecting.limb_name == BP_R_ARM ? "right arm" : "left arm"].</span>", "<span class='danger'>The needle <B>painfully</B> rips out of your [affecting.limb_name == BP_R_ARM ? "right arm" : "left arm"].</span>")
-		affecting.take_damage(brute = 5, damage_flags = DAM_SHARP)
-		attached = null
+	if(istype(W, /obj/item/reagent_containers))
+		if(beaker)
+			to_chat(user, "There is already a reagent container loaded!")
+			return
+		usr.drop_from_inventory(W, src)
+		beaker = W
+		usr.visible_message(SPAN_NOTICE("[usr] attaches \the [W] to \the [src]."), SPAN_NOTICE("You attach \the [W] to \the [src]."))
 		update_icon()
 		return
-
-	if(!attached.dna)
-		return
-
-	if(NOCLONE in attached.mutations)
-		return
-
-	if(attached.species.flags & NO_BLOOD)
+	if(istype(W, /obj/item/clothing/mask/breath))
+		if(is_type_in_list(W, mask_blacklist))
+			to_chat(usr, "\The [W] is incompatible with \the [src].")
+			return
+		if(breath_mask)
+			to_chat(usr, "There is already a mask installed.")
+			return
+		usr.drop_from_inventory(W, src)
+		breath_mask = W
+		usr.visible_message(SPAN_NOTICE("[usr] places \the [W] in \the [src]."), SPAN_NOTICE("You place \the [W] in \the [src]."))
+		update_icon()
 		return
-
-	// Give blood
-	if(mode)
-		if(beaker.reagents.total_volume > 0)
-			beaker.reagents.trans_to_mob(attached, transfer_amount, CHEM_BLOOD)
-			update_icon()
-
-		if(toggle_stop) // Automatically detaches if the blood volume is at 100%
-			if(beaker.reagents.has_reagent(/decl/reagent/blood) && attached.get_blood_volume() >= 100)
-				visible_message("[icon2html(src, viewers(get_turf(src)))] \The <b>[src]</b> flashes a warning light, disengaging from [attached] automatically!")
-				playsound(src, 'sound/machines/buzz-two.ogg', 50)
-				src.attached = null
-				src.update_icon()
-				blood_message_sent = FALSE
-				update_icon()
+	if(istype(W, /obj/item/tank))
+		if(is_type_in_list(W, tank_blacklist))
+			to_chat(usr, "\The [W] is incompatible with \the [src].")
+			return
+		if(tank)
+			to_chat(usr, "There is already a tank installed!")
+			return
+		if(istype(W, /obj/item/tank/phoron))
+			if(tipped)
```

Comment:
Why is the phoron tank, specifically, harder to add to tipped IVs + not securable? Is it supposed to be heavier or more awkward or something?

Reply:
It's mostly because of spriting issues. It clipped out of frame whenever I put the phoron on the correct position and looked really weird in turn